### PR TITLE
[DATA_MODEL] rename commande_fournisseur_dispatch into receptiondet_batch

### PR DESCRIPTION
**Pull request by altairis-tof**
follow https://github.com/Dolibarr/dolibarr/pull/26021

### CODE REVIEW

**Review by lmag — APPROVED** [review]
@eldy

**Comment by altairis-tof** [comment]
next steps will be the renaming of the CommandeFournisseurDispatch class and then the renaming of fk_commande and fk_commandefourndet fields

**Comment by fappels** [comment]
Does this mean classic dispatch without reception object will get deprecated?

**Comment by altairis-tof** [comment]
Hello @fappels , i dont think so; the classic dispatch will still work.

File: htdocs/install/mysql/migration/19.0.0-20.0.0.sql [hunk 1]
```diff
@@ -205,4 +203,7 @@ ALTER TABLE llx_c_tva ADD COLUMN type_vat smallint NOT NULL DEFAULT 0 AFTER fk_p
 
 ALTER TABLE llx_categorie ADD COLUMN position integer DEFAULT 0 AFTER color;
 
-ALTER TABLE llx_product DROP COLUMN onportal; 
+ALTER TABLE llx_product DROP COLUMN onportal;
+
+RENAME TABLE llx_commande_fournisseur_dispatch_extrafields TO llx_receptiondet_batch_extrafields;
```
Comment (eldy) [inline htdocs/install/mysql/migration/19.0.0-20.0.0.sql:208]: To rename a table, you must use "ALTER TABLE ... RENAME ..."  (se example at top of the file).

also you must update the trigger used for pgslq:

-- VPGSL8.2 CREATE TRIGGER update_customer_modtime BEFORE UPDATE ON llx_receptiondet_batch FOR EACH ROW EXECUTE PROCEDURE update_modified_column_tms();

Comment (altairis-tof) [reply htdocs/install/mysql/migration/19.0.0-20.0.0.sql:208]: hello @eldy it seems that RENAME TABLE is also good (https://dev.mysql.com/doc/refman/8.0/en/rename-table.html and https://popsql.com/learn-sql/mysql/how-to-rename-a-table-in-mysql) but maybe you have a good reason ti not use it ?

Comment (altairis-tof) [reply htdocs/install/mysql/migration/19.0.0-20.0.0.sql:210]: trigger updated here if i understood what you said : https://github.com/Dolibarr/dolibarr/pull/27528/commits/933a36cf1e257156b4b1448b349ee8df8521184d

Comment (altairis-tof) [reply htdocs/install/mysql/migration/19.0.0-20.0.0.sql]: also change the way to rename table https://github.com/Dolibarr/dolibarr/pull/27528/commits/c2345a48c3a915badb7675ca363e1cab6b271328

**Comment by altairis-tof** [comment]
@eldy fixes done and conflicts solved; i am eager to see this one merged and work on the next PR ;-)

**Review by lmag — COMMENTED** [review]
@eldy

**Comment by altairis-tof** [comment]
Hello @eldy i really need this PR to be merged before creating ReceptionDetBatch class and modify CommandeFournisseurDispatch to inherit from the new one. I will have too many conflicts otherwise.
(@lmag @FHenry )

**Comment by lmag** [comment]
@eldy it's ok for you ?

**Review by lmag — APPROVED** [review]

**Review by lmag — APPROVED** [review]

**Review by FHenry — APPROVED** [review]

File: htdocs/install/pgsql/functions/functions.sql [hunk 2]
```diff
@@ -159,6 +159,7 @@ CREATE TRIGGER update_customer_modtime BEFORE UPDATE ON llx_propal FOR EACH ROW
 CREATE TRIGGER update_customer_modtime BEFORE UPDATE ON llx_propal_extrafields FOR EACH ROW EXECUTE PROCEDURE update_modified_column_tms();
 CREATE TRIGGER update_customer_modtime BEFORE UPDATE ON llx_propal_merge_pdf_product FOR EACH ROW EXECUTE PROCEDURE update_modified_column_tms();
 CREATE TRIGGER update_customer_modtime BEFORE UPDATE ON llx_propaldet_extrafields FOR EACH ROW EXECUTE PROCEDURE update_modified_column_tms();
+CREATE TRIGGER update_customer_modtime BEFORE UPDATE ON llx_receptiondet_batch FOR EACH ROW EXECUTE PROCEDURE update_modified_column_tms();
```
Comment (FHenry) [inline htdocs/install/pgsql/functions/functions.sql:162]: No needed because duplicate of line 92

**Review by FHenry — CHANGES_REQUESTED** [review]
@altairis-tof There is duplicate create PGSql method in function.sql

**Pull request merged**